remote_config/src/ios/remote_config_ios.mm

```diff
@@ -491,9 +516,24 @@ static Variant FIRValueToVariant(FIRRemoteConfigValue *in_value, ValueInfo *valu
 }
 
 ConfigUpdateListenerRegistration* RemoteConfigInternal::AddOnConfigUpdateListener(
-      std::function<void(ConfigUpdate&&, RemoteConfigError)>
-          config_update_listener) {
-            return nullptr;
+      LambdaConfigUpdateListener<ConfigUpdate, RemoteConfigError> *config_update_listener) {
+    FIRConfigUpdateListenerRegistration *registration;
+    registration = [impl() addOnConfigUpdateListener: ^(FIRRemoteConfigUpdate *_Nullable update,
+                                  NSError *_Nullable error) {
+          if (error) {
+            config_update_listener->onError(ConvertFIRRemoteConfigUpdateError(error));
+          }
+          if (update) {
+            config_update_listener->onUpdate(ConvertConfigUpdateKeys(update.updatedKeys));
+          }
+    }];
+
+    ConfigUpdateListenerRegistration *registrationWrapper =
+      new ConfigUpdateListenerRegistration([&registration]() {
```

Comment:
If registration is a pointer, it may be fine to capture by value instead of reference (removing the &).
Since registration is a local variable I worry that capture by reference will result in the variable falling out of scope before this callback is evaluated.

This is probably another case of test it and see if it works as-is.

Reply:
Added listener removal test and it looks to work fine

Reply:
Alright, sounds good